services/structureddate/structureddate/src/main/java/org/collectionspace/services/structureddate/antlr/ANTLRStructuredDateEvaluator.java

```diff
@@ -460,12 +467,75 @@ public void exitInvStrDate(InvStrDateContext ctx) {
 		Integer dayOfMonth = (Integer) stack.pop();
 		Integer numMonth = (Integer) stack.pop();
 		Integer year = (Integer) stack.pop();
-		Era era = (Era) stack.pop();
+		Era era = (ctx.era() == null) ? null : (Era) stack.pop();
+
+		stack.push(year);
+		stack.push(numMonth);
+		stack.push(dayOfMonth);
+		stack.push(era);
+	}
+
+	@Override
+	public void exitDayFirstDate(DayFirstDateContext ctx) {
+		if (ctx.exception != null) return ;
+		
+		Era era = (ctx.era() == null) ? null : (Era) stack.pop();
+		Integer year = (Integer) stack.pop();
+		Integer month = (Integer) stack.pop();
+		Integer dayOfMonth = (Integer) stack.pop();
+
+		stack.push(year);
+		stack.push(month);
+		stack.push(dayOfMonth);
+		stack.push(era);
+	}
+
+	@Override
+	public void exitDayOrYearFirstDate(DayOrYearFirstDateContext ctx) {
+		if (ctx.exception != null) return;
+
+		Era era = (stack.size() == 3) ? null : (Era) stack.pop();
```

Comment:
Hi Ray, I already try doing this but it doesn't work correctly. Valid dates are parsed correctly but invalid ones like `100 June 2010` are not caught because they call the method multiple times. If i push a null era, then on the next call, `num2` becomes the null era, and it fails, causing another call. Im not sure why it doesn't fail at first when i check 
```java
if (dayOfMonth > 31 || dayOfMonth <= 0) {
   throw new StructuredDateFormatException("unexpected day of month '" + Integer.toString(dayOfMonth) + "'");
}
```
though I have tried printing inside, and it gets printed but it doesnt exit upon throwing the exception. I also tried just not passing in an era into the stack, but `exitDate` expects an era so perhaps I should move the era check there? I guess that might be a better place for it in case there are future dates without eras as well